### ISSUE TITLE
WLSDPLY-05017 when ForeignJMS ConnectionURL points to a directory and not to a file

### ISSUE THREAD

**Issue by flistell** [oracle/weblogic-deploy-tooling#726]
While running `discoverDomain.sh` on a domain  with a foreign JMS server configured, I'm getting these errors:

```
####<Sep 7, 2020 4:05:44 PM> <WARNING> <Validator> <__validate_attribute> <WLSDPLY-05017> <Expected value of the ConnectionURL attribute at location resources:/JMSSystemResource/app0JMSModule/JmsResource/ForeignServer/app0ForeignServer to be a string data type, but it was a <type 'NoneType'>>
####<Sep 7, 2020 4:05:44 PM> <SEVERE> <Validator> <__validate_path_tokens_attribute> <WLSDPLY-05023> <Value of attribute ConnectionURL at model location resources:/JMSSystemResource/app0JMSModule/JmsResource/ForeignServer/app0ForeignServer has an unexpected data type: <type 'NoneType'>>
    ...
 5. WLSDPLY-05017: Expected value of the ConnectionURL attribute at location resources:/JMSSystemResource/app0JMSModule/JmsResource/ForeignServer/app0ForeignServer to be a string data type, but it was a <type 'NoneType'>
```

Note that `DOMIAN_HOME/config/jms/app0jmsmodule-jms.xml` has `connection-url` element correctly set:

```xml
        <connection-url>file:///var/wls/app0/mq</connection-url>
```

I'm using WDT 1.9.5

**Comment by flistell** [comment]
My analysis of the issue.
Out Forign JMS provider is a IBM MQ. IBM MQ client requires to specify "ConnectionURL" as a directory that would contain a `.bindings` file specifying the connectoin properties for the MQ client libs.

The discoverDomain logs show:
```

####<Sep 7, 2020 4:34:14 PM> <FINER> <WLSDeployArchive> <validateExistingFile> <> <ENTRY /var/wls/app0/mq configFile /domains/domain1_wdt_archive.zip addForeignServerFile false>
####<Sep 7, 2020 4:34:14 PM> <FINER> <WLSDeployArchive> <validateExistingFile> <> <THROW
java.lang.IllegalArgumentException: Unable to add file to the archive /domains/domain1_wdt_archive.zip because the file /var/wls/app0/mq is a directory
        at oracle.weblogic.deploy.util.WLSDeployArchive.validateExistingFile(WLSDeployArchive.java:1450)
        at oracle.weblogic.deploy.util.WLSDeployArchive.validateExistingFile(WLSDeployArchive.java:1430)
        at oracle.weblogic.deploy.util.WLSDeployArchive.addForeignServerFile(WLSDeployArchive.java:1000)
        at ...
####<Sep 7, 2020 4:34:14 PM> <WARNING> <PyReflectedFunction> <__call__> <WLSDPLY-06493> <Added Foreign Server app0ForeignServer bindings file from location /var/wls/app0/mq to archive file with name Unable to add file to the archive /domains/domain1_wdt_archive.zip because the file /var/wls/app0/mq is a directory>
####<Sep 7, 2020 4:34:14 PM> <FINER> <JmsResourcesDiscoverer> <_add_foreign_server_binding> <> <RETURN>
```

A good solution would be to create an archive with the path specified by "ConnectionURL" in case of ForeignJMS and then copy it to the archive. This would work with both files and directories.

**Comment by ddsharpe** [comment]
@flistell testing with IBM MQ is not in the test suite, so I wouldn't be surprised if this is a bug.  We have a few other priorities in the queue but will check into this as soon as we can.

**Comment by flistell** [comment]
More info: that directory usually contains a single file called '.bindings'.

**Comment by CarolynRountree** [comment]
Fixed by PR #728
